crates/types/src/config/common.rs

```diff
@@ -506,6 +533,14 @@ pub enum MetadataStoreClient {
         #[cfg_attr(feature = "schemars", schemars(with = "String"))]
         addresses: Vec<String>,
     },
+    /// Uses an object store as a metadata store.
+    ObjectStore {
+        credentials: ObjectStoreCredentials,
+
+        /// # The bucket name to use for storage
+        #[cfg_attr(feature = "schemars", schemars(with = "String"))]
```

Comment:
You probably don't have to add this given that it's an identity function (String -> String).